Fix portsmplexer build

- Program: Recursor
 - Issue type: Bug report

### Short description
This recent [commit](https://github.com/PowerDNS/pdns/commit/7c58a81f724d0d939d53a7f9599bb7db47f7c3a4) updated portsmplexer, which now fails to compile.

### Environment
 - Operating system: SmartOS/illumos
 - Software version: 4.7.3
 - Software source: [pkgsrc](https://github.com/NetBSD/pkgsrc/tree/trunk/net/powerdns-recursor)

### Steps to reproduce
Simply build 4.7.3 on any Solaris or illumos distribution.

### Expected behaviour
A successful build.

### Actual behaviour
There are two errors in the above commit:

* The signature of `PortsFDMultiplexer()` changed but not all call-sites were updated, causing:

```
portsmplexer.cc: In function 'FDMultiplexer* makePorts(unsigned int)':                                                                        portsmplexer.cc:44:33: error: no matching function for call to 'PortsFDMultiplexer::PortsFDMultiplexer()'                                     
   44 |   return new PortsFDMultiplexer();                                                                                                    
      |                                 ^                                                                                                     
portsmplexer.cc:20:3: note: candidate: 'PortsFDMultiplexer::PortsFDMultiplexer(unsigned int)'                                                 
   20 |   PortsFDMultiplexer(unsigned int maxEventsHint);                                                                                     
      |   ^~~~~~~~~~~~~~~~~~                                           
portsmplexer.cc:20:3: note:   candidate expects 1 argument, 0 provided                                                                        
```

* Switching to `d_pevents.size()` causes a mismatch in arguments passed to `std::min()`:

```
portsmplexer.cc: In member function 'virtual void PortsFDMultiplexer::getAvailableFDs(std::vector<int>&, int)':                               
portsmplexer.cc:97:86: error: no matching function for call to 'min(int, std::vector<port_event>::size_type)'                                 
   97 |   int ret = port_getn(d_portfd, d_pevents.data(), min(PORT_MAX_LIST, d_pevents.size()), &numevents, &timeoutspec);                    
      |                                                                                      ^                         
In file included from /opt/local/gcc10/include/c++/10.3.0/algorithm:62,                                                                       
                 from /home/pbulk/build/net/powerdns-recursor/work/.buildlink/include/boost/multi_index_container.hpp:19,                     
                 from mplexer.hh:24,                                                                                                          
                 from portsmplexer.cc:9:                                                                                                      
/opt/local/gcc10/include/c++/10.3.0/bits/stl_algo.h:3479:5: note: candidate: 'template<class _Tp, class _Compare> constexpr _Tp std::min(std::
initializer_list<_Tp>, _Compare)'                                                                                                             
 3479 |     min(initializer_list<_Tp> __l, _Compare __comp)            
      |     ^~~                                                                                                                               

portsmplexer.cc: In member function 'virtual int PortsFDMultiplexer::run(timeval*, int)':
portsmplexer.cc:158:86: error: no matching function for call to 'min(int, std::vector<port_event>::size_type)'
  158 |   int ret = port_getn(d_portfd, d_pevents.data(), min(PORT_MAX_LIST, d_pevents.size()), &numevents, &timeoutspec);
      |                                                                                      ^
In file included from /opt/local/gcc10/include/c++/10.3.0/algorithm:62,
                 from /home/pbulk/build/net/powerdns-recursor/work/.buildlink/include/boost/multi_index_container.hpp:19,
                 from mplexer.hh:24,
                 from portsmplexer.cc:9:
/opt/local/gcc10/include/c++/10.3.0/bits/stl_algo.h:3479:5: note: candidate: 'template<class _Tp, class _Compare> constexpr _Tp std::min(std::
initializer_list<_Tp>, _Compare)'
 3479 |     min(initializer_list<_Tp> __l, _Compare __comp)
      |     ^~~
```


### Other information
I've committed a fix for this issue here:

https://github.com/NetBSD/pkgsrc/commit/9b64f206908b219c2a6177bf138e30513498571f#diff-0509e56dfedfb86c0d5db234a834497010203fe425ff718298eed33375ff7272

You are more than welcome to take the patch.

--- a/pdns/portsmplexer.cc
+++ b/pdns/portsmplexer.cc
@@ -39,9 +39,9 @@ private:
   std::vector<port_event_t> d_pevents;
 };
 
-static FDMultiplexer* makePorts(unsigned int)
+static FDMultiplexer* makePorts(unsigned int maxEventsHint)
 {
-  return new PortsFDMultiplexer();
+  return new PortsFDMultiplexer(maxEventsHint);
 }
 
 static struct PortsRegisterOurselves
@@ -94,7 +94,7 @@ void PortsFDMultiplexer::getAvailableFDs(std::vector<int>& fds, int timeout)
   timeoutspec.tv_sec = timeout / 1000;
   timeoutspec.tv_nsec = (timeout % 1000) * 1000000;
   unsigned int numevents = 1;
-  int ret = port_getn(d_portfd, d_pevents.data(), min(PORT_MAX_LIST, d_pevents.size()), &numevents, &timeoutspec);
+  int ret = port_getn(d_portfd, d_pevents.data(), min(PORT_MAX_LIST, static_cast<int>(d_pevents.size())), &numevents, &timeoutspec);
 
   /* port_getn has an unusual API - (ret == -1, errno == ETIME) can
      mean partial success; you must check (*numevents) in this case
@@ -155,7 +155,7 @@ int PortsFDMultiplexer::run(struct timeval* now, int timeout)
   timeoutspec.tv_sec = timeout / 1000;
   timeoutspec.tv_nsec = (timeout % 1000) * 1000000;
   unsigned int numevents = 1;
-  int ret = port_getn(d_portfd, d_pevents.data(), min(PORT_MAX_LIST, d_pevents.size()), &numevents, &timeoutspec);
+  int ret = port_getn(d_portfd, d_pevents.data(), min(PORT_MAX_LIST, static_cast<int>(d_pevents.size())), &numevents, &timeoutspec);
 
   /* port_getn has an unusual API - (ret == -1, errno == ETIME) can
      mean partial success; you must check (*numevents) in this case

